contracts/beacon/Voucher.sol

```diff
@@ -164,11 +184,69 @@ contract Voucher is OwnableUpgradeable, IVoucher {
             workerpoolOrder,
             requestOrder
         );
-        $._sponsoredAmounts[dealId] = sponsoredAmount;
+        $._voucherMatchedDeals[dealId].exists = true;
+        $._voucherMatchedDeals[dealId].sponsoredAmount = uint64(sponsoredAmount);
         emit OrdersBoostMatchedWithVoucher(dealId);
         return dealId;
     }
 
+    function claim(bytes32 dealId, uint256 index) external {
+        VoucherStorage storage $ = _getVoucherStorage();
+        IVoucherHub voucherHub = IVoucherHub($._voucherHub);
+        address iexecPoco = voucherHub.getIexecPoco();
+        // Check if task is already claimed.
+        bytes32 taskId = keccak256(abi.encodePacked(dealId, index));
+        require(!$._claimedTasks[taskId], "Voucher: task already claimed");
```

Comment:
why do we want to check that, it won't revert if the task is already claim in Poco directky

Reply:
The check is to make sure the task is not claimed twice on the voucher.